src/gui/MainWindow.h

```diff
@@ -55,6 +55,7 @@ class MainWindow : public QMainWindow, public Ui::MainWindow, public InputEventH
   SourceFile *parsed_file; // Last parse for include list
   std::shared_ptr<AbstractNode> absolute_root_node; // Result of tree evaluation
   std::shared_ptr<AbstractNode> root_node; // Root if the root modifier (!) is used
+  static int python_active;					
```

Comment:
Why static? You're checking this for each MainWindow instance, so it should probably be a regular instance member.

Reply:
I assume it's there to prevent any concurrent access, across multiple MainWindows. I have not checked but it might not be needed as we already have that annoying global lock via `GuiLocker::gui_locked`.